### PR TITLE
Error overlay added and triggered where metadata tables empty.

### PR DESCRIPTION
**Pull request by thmasjs**
# Changes

*Describe the changes made, including any changes to tests, dependencies or docs. Link to any relevant issues below.*

- Closes #XXXX *(replace XXXX with GitHub issue number)*
- ...

# Tasks

- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Has documentation (including user guide) been updated
- [ ] Have new dependencies been added (or changed) in relevant `requirements.txt`
- [ ] Code linting, tests and other workflows pass

### CODE REVIEW

**Comment by thmasjs** [comment]
Simple test: find a page that has filters based off empty metadata tables. This page should now show: 
image

Component is fully dynamic meaning we can send any error through for future use.

**Review by brereh — CHANGES_REQUESTED** [review]
Need to tighten export/imports to be consistent with rest of app.

**Comment by thmasjs** [comment]
Have changed it to use ErrorProvider and Context, might need some tweaks as I rushed it a bit and basically copied mapContext/reducer. but it should work as intended.

Apps will need a change to cover the app in errorprovider:

`  return (
    <div className="App">
      <AuthProvider>
        <ErrorProvider>
        <ThemeProvider theme={theme}>
        <AppContext.Provider value={appConfig}>
          <Navbar />
          <Dashboard>
            <Routes>
              <Route path="/login" element={<Login />} />
              <Route path="/unauthorized" element={<Unauthorized />} />
              <Route path="/" element={<HomePageWithRoleValidation />} />
              {appConfig.appPages.map((page) => {
                const PageComponent = isAuthRequired
                  ? withRoleValidation(PageSwitch)
                  : PageSwitch;
                const WrappedPageComponent =
                  composeHOCs(withWarning, withTermsOfUse)(PageComponent);
                return (
                  <Route
                    key={page.pageName}
                    path={page.url}
                    element={<WrappedPageComponent pageConfig={page} />}
                  />
                );
              })}
              {/* Catch-all for undefined routes */}
              <Route path="*" element={<NotFoundWithRoleValidation />} />
            </Routes>
          </Dashboard>
        </AppContext.Provider>
        </ThemeProvider>
        </ErrorProvider>
      </AuthProvider>
    </div>
  );`

File: packages/vis-core/src/contexts/MapContext.jsx [hunk 1]
```diff
@@ -47,6 +47,8 @@ export const MapProvider = ({ children }) => {
   const appContext = useContext(AppContext);
   const pageContext = useContext(PageContext);
   const { dispatch: filterDispatch } = useContext(FilterContext);
+  const _errorCtx = useContext(ErrorContext);
```
Comment (brereh) [inline packages/vis-core/src/contexts/MapContext.jsx]: As in hook, destructure to errorState and errorDispatch in one line.

File: packages/vis-core/src/Components/ErrorOverlay/ErrorOverlay.jsx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+          {showTechnicalDetails && technicalDetails && (
+            <CollapsibleSection>
+              <CollapsibleHeader 
+                onClick={toggleDetails}
+                aria-expanded={isDetailsOpen}
+                aria-controls="error-details"
+              >
+                <span>Technical Details</span>
+                <ChevronIcon $isOpen={isDetailsOpen}>▼</ChevronIcon>
```
Comment (brereh) [inline packages/vis-core/src/Components/ErrorOverlay/ErrorOverlay.jsx]: Use chevron styling consistent with rest of app (heroicons)

File: packages/vis-core/src/hooks/useFetchVisualisationData.jsx [hunk 3]
```diff
@@ -268,6 +282,8 @@ export const useFetchVisualisationData = (
     visualisation?.queryParams,
     visualisation?.pathParams,
     fetchDataForVisualisation,
+    errorState,
```
Comment (brereh) [inline packages/vis-core/src/hooks/useFetchVisualisationData.jsx]:
```suggestion
```
Remove errorState from dependency array.
